### ISSUE TITLE
Error in post-command-hook (yas--post-command-handler): (wrong-type-argument markerp 18)

### ISSUE THREAD

**Issue by netjune** [joaotavora/yasnippet#1032]
Reproduction steps:
1. emacs -Q
2. install packages from melpa: yasnippet, go-mode
3. open an empty go file, turn on yasnippet in the buffer, and load two snippets: sel, ch
4. input "sel", then M-x yas-expand
5. in the snippet field, input "ch", then M-x yas-expand
6. input "qqq". emacs display error message like this:

`Error in post-command-hook (yas--post-command-handler): (wrong-type-argument markerp 18)`

the sel snippet:
```
# name: select
# key: sel
# --
select {
case ${1:condition}:
	$0
}
```
the ch snippet:
```
# name: <-ch
# key: ch
# --
<-${1:ch}
```

the backstrace:
```
Debugger entered--Lisp error: (wrong-type-argument markerp 18)
  yas--advance-end-maybe(#s(yas--field :number 1 :start 15 :end 18 :parent-field nil :mirrors nil :transform nil :modified-p t :next #s(yas--exit :marker 22 :next nil)) 20)
  yas--commit-snippet(#s(yas--snippet :expand-env nil :fields (#s(yas--field :number 1 :start #<marker at 17 in a.go> :end #<marker at 19 in a.go> :parent-field nil :mirrors nil :transform nil :modified-p t :next nil)) :exit nil :id 9 :control-overlay nil :active-field #s(yas--field :number 1 :start #<marker at 17 in a.go> :end #<marker at 19 in a.go> :parent-field nil :mirrors nil :transform nil :modified-p t :next nil) :previous-active-field #s(yas--field :number 1 :start 15 :end 18 :parent-field nil :mirrors nil :transform nil :modified-p t :next #s(yas--exit :marker 22 :next nil)) :force-exit nil))
  #f(compiled-function () #<bytecode 0x4160fe01>)()
  funcall(#f(compiled-function () #<bytecode 0x4160fe01>))
  (let nil (funcall (quote #f(compiled-function () #<bytecode 0x4160fe01>))))
  eval((let nil (funcall (quote #f(compiled-function () #<bytecode 0x4160fe01>)))))
  yas--check-commit-snippet()
  yas--post-command-handler()
```

**Comment by netjune** [comment]
Another test case:

I set yas-verbose to 4 and add code to print the active snippets in yas--message:
```
(defun yas--message (level message &rest args)
  "When LEVEL is at or below `yas-verbosity', log MESSAGE and ARGS."
  (when (>= yas-verbosity level)
    (message "%s" (apply #'yas--format message args))
    (message "--- active snippets: %s\n" yas--active-snippets)
    ))
```

There are four nested snippets. The log shows that one middle snippet(the snippet id: 1) leaked. 

The log content:
```
[yas] snippet 0 expanded.
--- active snippets: (#s(yas--snippet nil (#s(yas--field 1 #<marker at 14989 in device_center.go> #<marker at 14998 in device_center.go> nil nil nil nil #s(yas--exit #<marker at 15002 in device_center.go> nil))) #s(yas--exit #<marker at 15002 in device_center.go> nil) 0 #<overlay from 14974 to 15005 in device_center.go> #s(yas--field 1 #<marker at 14989 in device_center.go> #<marker at 14998 in device_center.go> nil nil nil nil #s(yas--exit #<marker at 15002 in device_center.go> nil)) nil nil))

[yas] snippet 1 expanded.
--- active snippets: (#s(yas--snippet nil (#s(yas--field 1 #<marker at 14991 in device_center.go> #<marker at 14993 in device_center.go> nil nil nil nil nil)) nil 1 #<overlay from 14989 to 14993 in device_center.go> #s(yas--field 1 #<marker at 14991 in device_center.go> #<marker at 14993 in device_center.go> nil nil nil nil nil) #s(yas--field 1 #<marker at 14989 in device_center.go> #<marker at 14993 in device_center.go> nil nil nil t #s(yas--exit #<marker at 14997 in device_center.go> nil)) nil) #s(yas--snippet nil (#s(yas--field 1 #<marker at 14989 in device_center.go> #<marker at 14993 in device_center.go> nil nil nil t #s(yas--exit #<marker at 14997 in device_center.go> nil))) #s(yas--exit #<marker at 14997 in device_center.go> nil) 0 #<overlay from 14974 to 15000 in device_center.go> #s(yas--field 1 #<marker at 14989 in device_center.go> #<marker at 14993 in device_center.go> nil nil nil t #s(yas--exit #<marker at 14997 in device_center.go> nil)) nil nil))

[yas] snippet 2 expanded.
--- active snippets: (#s(yas--snippet nil nil nil 2 #<overlay from 14996 to 15010 in device_center.go> nil #s(yas--field 1 #<marker at 14991 in device_center.go> #<marker at 15010 in device_center.go> nil nil nil t nil) t) #s(yas--snippet nil (#s(yas--field 1 #<marker at 14991 in device_center.go> #<marker at 15010 in device_center.go> nil nil nil t nil)) nil 1 #<overlay from 14989 to 15010 in device_center.go> #s(yas--field 1 #<marker at 14991 in device_center.go> #<marker at 15010 in device_center.go> nil nil nil t nil) #s(yas--field 1 #<marker at 14989 in device_center.go> #<marker at 14996 in device_center.go> nil nil nil t #s(yas--exit #<marker at 15014 in device_center.go> nil)) nil) #s(yas--snippet nil (#s(yas--field 1 #<marker at 14989 in device_center.go> #<marker at 14996 in device_center.go> nil nil nil t #s(yas--exit #<marker at 15014 in device_center.go> nil))) #s(yas--exit #<marker at 15014 in device_center.go> nil) 0 #<overlay from 14974 to 15017 in device_center.go> #s(yas--field 1 #<marker at 14989 in device_center.go> #<marker at 14996 in device_center.go> nil nil nil t #s(yas--exit #<marker at 15014 in device_center.go> nil)) nil nil))

[yas] Snippet 2 exited.
--- active snippets: (#s(yas--snippet nil (#s(yas--field 1 #<marker at 14991 in device_center.go> #<marker at 15010 in device_center.go> nil nil nil t nil)) nil 1 #<overlay from 14989 to 15010 in device_center.go> #s(yas--field 1 #<marker at 14991 in device_center.go> #<marker at 15010 in device_center.go> nil nil nil t nil) #s(yas--field 1 #<marker at 14989 in device_center.go> #<marker at 14996 in device_center.go> nil nil nil t #s(yas--exit #<marker at 15014 in device_center.go> nil)) nil) #s(yas--snippet nil (#s(yas--field 1 #<marker at 14989 in device_center.go> #<marker at 14996 in device_center.go> nil nil nil t #s(yas--exit #<marker at 15014 in device_center.go> nil))) #s(yas--exit #<marker at 15014 in device_center.go> nil) 0 #<overlay from 14974 to 15017 in device_center.go> #s(yas--field 1 #<marker at 14989 in device_center.go> #<marker at 14996 in device_center.go> nil nil nil t #s(yas--exit #<marker at 15014 in device_center.go> nil)) nil nil))

[yas] Snippet 0 exited.
--- active snippets: (#s(yas--snippet nil (#s(yas--field 1 #<marker at 14991 in device_center.go> #<marker at 15010 in device_center.go> nil nil nil t nil)) nil 1 #<overlay from 14989 to 15010 in device_center.go> #s(yas--field 1 #<marker at 14991 in device_center.go> #<marker at 15010 in device_center.go> nil nil nil t nil) #s(yas--field 1 14989 14996 nil nil nil t #s(yas--exit 15014 nil)) nil))

Error in post-command-hook (yas--post-command-handler): (wrong-type-argument markerp 14996)
```

**Comment by netjune** [comment]
There are two issues here. One is in floor 1. One is in floor 2. They are different.

The first one has been fixed in 0.14.0. The second exists in 0.14.0.

I will close this one and open another issue for the second.
